platform/ui/src/components/SidePanel/SidePanel.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { Button, Icon } from '@ohif/ui';
+
+const borderSize = 4;
+const expandedWidth = 320;
+const collapsedWidth = 60;
+
+const baseStyle = {
+  maxWidth: `${expandedWidth}px`,
+  width: `${expandedWidth}px`,
+};
+
+const collapsedHideWidth = expandedWidth - collapsedWidth - borderSize;
+const styleMap = {
+  open: {
+    left: { marginLeft: '0px' },
+    right: { marginRight: '0px' },
+  },
+  closed: {
+    left: { marginLeft: `-${collapsedHideWidth}px` },
+    right: { marginRight: `-${collapsedHideWidth}px` },
+  },
+};
+
+const baseClasses =
+  'transition-all duration-300 ease-in-out h-100 bg-primary-dark border-black flex flex-col justify-start';
+
+const classesMap = {
+  open: {
+    left: `border-r-${borderSize}`,
+    right: `border-l-${borderSize}`,
+  },
+  closed: {
+    left: `border-r-${borderSize} items-end`,
+    right: `border-l-${borderSize} items-start`,
+  },
+};
+
+const openIconName = {
+  left: 'panel-left',
+  right: 'panel-right',
+};
+
+const SidePanel = ({
+  side,
+  className,
+  children,
+  defaultIsOpen,
```

Comment:
`defaultIsOpen` I think is not a good name, we can rename it to `isOpen` or just `open`. It's clearer to have a prop like this than with a `default` prefix. I saw you maybe did it because there's a `useState` using `isOpen`. Maybe you can rename the prop and rename to defaultIsOpen when getting the variable here.

```suggestion
  isOpen: defaultIsOpen,
```

What do you think?

Reply:
I thought about that and changed to its current name, in my idea, once we set it to isOpen or Open, gives the user the idea that the parent is controling.. and right now, its not. thats why I changed it to defaultIsOpen. but we can rename it, just don't think open or isOpen good either

Reply:
Hmm, It's true! Sorry, I hadn't noticed. We can leave it like this and think later if the open/closed state will be controlled in the component.